tests/test_start_date.py

```diff
@@ -0,0 +1,135 @@
+        ### First Sync
+        ##########################################################################
+
+        # instantiate connection
+        conn_id_1 = connections.ensure_connection(self)
+
+        # Note: test data not available for following streams: metadata, email_send_skip, email_complaint, email_click
+        streams_to_test = self.expected_streams()
+
+        # run check mode
+        found_catalogs_1 = self.run_and_verify_check_mode(conn_id_1)
+
+        # table and field selection
+        test_catalogs_1_all_fields = [catalog for catalog in found_catalogs_1
+                                      if catalog.get('tap_stream_id') in streams_to_test]
+        self.perform_and_verify_table_and_field_selection(
+            conn_id_1, test_catalogs_1_all_fields, select_all_fields=True)
+
+        # run initial sync
+        record_count_by_stream_1 = self.run_and_verify_sync(conn_id_1)
+        synced_records_1 = runner.get_records_from_target_output()
+
+        ##########################################################################
+        ### Update START DATE Between Syncs
+        ##########################################################################
+
+        LOGGER.info(f"REPLICATION START DATE CHANGE: {self.START_DATE} ===>>> {self.start_date_2}")
+        self.START_DATE = self.start_date_2
+
+        ##########################################################################
+        ### Second Sync
+        ##########################################################################
+
+        # create a new connection with the new start_date
+        conn_id_2 = connections.ensure_connection(self, original_properties=False)
+
+        # run check mode
+        found_catalogs_2 = self.run_and_verify_check_mode(conn_id_2)
+
+        # table and field selection
+        test_catalogs_2_all_fields = [catalog for catalog in found_catalogs_2
+                                      if catalog.get('tap_stream_id') in streams_to_test]
+        self.perform_and_verify_table_and_field_selection(
+            conn_id_2, test_catalogs_2_all_fields, select_all_fields=True)
+
+        # run sync
+        record_count_by_stream_2 = self.run_and_verify_sync(conn_id_2)
+        synced_records_2 = runner.get_records_from_target_output()
+
+        for stream in streams_to_test:
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_start_date_1 = self.start_date_1['start_date']
+                expected_start_date_2 = self.start_date_2
+
+                # All the streams obey the start date
+                # collect information for assertions from sync 1 and sync 2 based on expected values
+                record_count_sync_1 = record_count_by_stream_1.get(stream, 0)
+                record_count_sync_2 = record_count_by_stream_2.get(stream, 0)
+
```

Comment:
Verify stream has records in each sync.

Reply:
Fixed this at,

- [First sync assertion](https://github.com/singer-io/tap-iterable/blob/TDL-21942-add-tap-tester-tests/tests/test_start_date.py#L56-#L60)
- [Second sync assertion](https://github.com/singer-io/tap-iterable/blob/TDL-21942-add-tap-tester-tests/tests/test_start_date.py#L93-#L95)